src/libstd/ffi/c_str.rs

```diff
@@ -266,11 +266,17 @@ impl CString {
 
     /// Transfers ownership of the string to a C caller.
     ///
-    /// The pointer must be returned to Rust and reconstituted using
-    /// `from_raw` to be properly deallocated. Specifically, one
+    /// Unless the same memory allocator is used for both Rust and C,
+    /// the acquired pointer must be returned to Rust and reconstituted
+    /// using `from_raw` to be properly deallocated. Specifically, one
```

Comment:
Indeed, this would logically extend to from_raw. It's not a usecase I needed yet, so I haven't considered it here. Happy to work on the phrasing if people are on board with the general idea!